### DISCUSSION TITLE
Automatic time step with Electrostatic solver?

### DISCUSSION THREAD

**Discussion by BCatGA** [BLAST-WarpX/warpx#5431] [Simulation Inputs] [answered]
Hi,

Is it possible to automatically calculate the time step when using the Electrostatic solver? If I don't define time_step_size in my picmi.Simulation command, I get the following error:

> ERROR   : warpx.const_dt must be specified with the electrostatic solver, or
>                 warpx.dt_update_interval must be > 0.

The online documentation for the Electrostatic solver (https://warpx.readthedocs.io/en/latest/usage/python.html) says 

> warpx_dt_update_interval (string, optional (default = -1)) – How frequently the timestep is updated. Adaptive timestepping is disabled when this is <= 0.

But if I add "warpx_dt_update_interval=1" to the argument for Electrostatic solver, WarpX doesn't recognize it:

> Unexpected keyword argument: "['warpx_dt_update_interval']"

Looking at the picmi.py file, it seems like the keyword is actually "dt_update_interval":

>         self.dt_update_interval = kw.pop("dt_update_interval", None)

But if I change "warpx_dt_update_interval=1" to "dt_update_interval=1", WarpX still doesn't recognize it:

> Unexpected keyword argument: "['dt_update_interval']"

Is there a way to get this to work, or am I stuck with a fixed time step?

Thanks for your help!

**Comment by dpgrote** [comment, marked as answer]
Thanks for the issue. This is a mistake in the picmi interface. I'll submit a PR to fix this. You can fix this temporarily by changing the line in picmi.py to `self.dt_update_interval = kw.pop("warpx_dt_update_interval", None)` and then pass in the keyword argument 'warpx_dt_update_interval`.

The fix is in PR #5434
